tornjak-frontend/src/components/dashboard/dashboard-details.js

```diff
@@ -0,0 +1,223 @@
+            } else if (clickedDashboardTable === "entriesdetails") {
+                //to filter agents in entriesdetails
+                selectedDataKey["agentsFilter"] = selectedData.parentId;
+                //to filter clusters in entriesdetails
+                selectedDataKey["clustersFilter"] = selectedData.clusterName;
+            } else if (clickedDashboardTable === "agentsdetails") {
+                //to filter clusters in agentsdetails
+                selectedDataKey["clustersFilter"] = selectedData.clusterName;
+                //to filter entries in agentsdetails
+                selectedDataKey["entriesFilter"] = selectedData.spiffeid;
+            }
+        }
+        return selectedDataKey;
+    }
+
+    render() {
+        const { classes, selectedData } = this.props;
+        return (
+            <div className={classes.root}>
+                <DashboardDrawer />
+                <main className={classes.content}>
+                    <div className={classes.appBarSpacer} />
+                    {(selectedData.length !== 0) && (!selectedData[0]) &&
+                        <div>
+                            {(this.props.globalClickedDashboardTable === "clustersdetails") &&
+                                <div className="clustersdetails">
+                                    <Container maxWidth="lg" className={classes.container}>
+                                        <Grid item xs={12}>
+                                            <Paper className={classes.paper}>
+                                                <p className={classes.detailsTitle}>Cluster Name : <b>{selectedData.name}</b></p>
+                                                <p className={classes.metadataTag}>Metadata</p>
+                                                <hr className={classes.dashboardDetalsLine}></hr>
+                                                <p className={classes.metadataDetails}>Created : <b>{selectedData.created} </b></p>
+                                                <p className={classes.metadataDetails}>Number of Nodes : <b>{selectedData.numNodes} </b></p>
+                                                <p className={classes.metadataDetails}>Number of Entries: <b>{selectedData.numEntries}</b> </p>
+                                            </Paper>
+                                        </Grid>
+                                    </Container>
+                                    <Container maxWidth="lg" className={classes.container}>
+                                        {/* Agents Table */}
+                                        <Grid item xs={12}>
+                                            <Paper className={classes.paper}>
+                                                <AgentsTable
+                                                    numRows={100}
+                                                    selectedDataKey={this.selectedDataKey()} />
```

Comment:
let's pass the actual value for the particular instance instead, so passing `selectedDataKey["clustersFilter"]` here instead, so each of the different tables don't need to know the special keyword `"clustersFilter"`